src/index.ts

```diff
@@ -146,33 +148,26 @@ export class Client {
     return new WriteTransaction(client, storeID, modelName)
   }
 
-  public async listen(
-    storeID: string,
-    modelName: string,
-    entityID: string,
-    callback: (reply: ListenReply.AsObject) => void,
-  ) {
-    return new Promise((resolve, reject) => {
-      const req = new ListenRequest()
-      req.setStoreid(storeID)
-      req.setModelname(modelName)
-      req.setEntityid(entityID)
-      const client = grpc.client(API.Listen, {
-        host: this.host,
-      }) as grpc.Client<ListenRequest, ListenReply>
-      client.onMessage((message: ListenReply) => {
-        callback(message.toObject())
-      })
-      client.onEnd((status: grpc.Code, message: string) => {
-        if (status !== grpc.Code.OK) {
-          reject(new Error(message))
-        } else {
-          resolve()
-        }
-      })
-      client.start()
-      client.send(req)
+  public listen(storeID: string, modelName: string, entityID: string, callback: (reply: ListenReply.AsObject) => void) {
+    const req = new ListenRequest()
+    req.setStoreid(storeID)
+    req.setModelname(modelName)
+    req.setEntityid(entityID)
+    const client = grpc.client(API.Listen, {
+      host: this.host,
+    }) as grpc.Client<ListenRequest, ListenReply>
+    client.onMessage((message: ListenReply) => {
+      callback(message.toObject())
+    })
+    client.onEnd((status: grpc.Code, message: string) => {
```

Comment:
Since this is no longer returning a promise, we only throw an error if it was prematurely ended. We could alternatively make this an async generator function, and let callers simply consume the contents until they end (then we'd have to cleanup) or it ends?

Reply:
This looks good to me... let's play with your new method and see how it feels.